dts/common/nordic/nrf54l20.dtsi

```diff
@@ -0,0 +1,614 @@
+				status = "disabled";
+				port = <1>;
+				gpiote-instance = <&gpiote20>;
+			};
+
+			gpiote20: gpiote@da000 {
+				compatible = "nordic,nrf-gpiote";
+				reg = <0xda000 0x1000>;
+				status = "disabled";
+				instance = <20>;
+			};
+
+			qdec20: qdec@e0000 {
+				compatible = "nordic,nrf-qdec";
+				reg = <0xe0000 0x1000>;
+				interrupts = <224 NRF_DEFAULT_IRQ_PRIORITY>;
+				status = "disabled";
+			};
+
+			qdec21: qdec@e1000 {
+				compatible = "nordic,nrf-qdec";
+				reg = <0xe1000 0x1000>;
+				interrupts = <225 NRF_DEFAULT_IRQ_PRIORITY>;
+				status = "disabled";
+			};
+
+			grtc: grtc@e2000 {
+				compatible = "nordic,nrf-grtc";
+				reg = <0xe2000 0x1000>;
+				cc-num = <12>;
+				status = "disabled";
+			};
+
+			dppic30: dppic@102000 {
+				compatible = "nordic,nrf-dppic";
+				reg = <0x102000 0x808>;
+				status = "disabled";
+			};
+
+			i2c30: i2c@104000 {
+				compatible = "nordic,nrf-twim";
+				#address-cells = <1>;
+				#size-cells = <0>;
+				reg = <0x104000 0x1000>;
+				clock-frequency = <I2C_BITRATE_STANDARD>;
+				interrupts = <260 NRF_DEFAULT_IRQ_PRIORITY>;
+				easydma-maxcnt-bits = <16>;
+				status = "disabled";
+			};
+
+			spi30: spi@104000 {
+				/*
+				 * This spi node can be either SPIM or SPIS,
+				 * for the user to pick:
+				 * compatible = "nordic,nrf-spim" or
+				 *              "nordic,nrf-spis".
+				 */
+				compatible = "nordic,nrf-spim";
+				#address-cells = <1>;
+				#size-cells = <0>;
+				reg = <0x104000 0x1000>;
+				interrupts = <260 NRF_DEFAULT_IRQ_PRIORITY>;
+				max-frequency = <DT_FREQ_M(8)>;
+				easydma-maxcnt-bits = <16>;
+				rx-delay-supported;
+				rx-delay = <1>;
+				status = "disabled";
+			};
+
+			uart30: uart@104000 {
+				compatible = "nordic,nrf-uarte";
+				reg = <0x104000 0x1000>;
+				interrupts = <260 NRF_DEFAULT_IRQ_PRIORITY>;
+				status = "disabled";
+			};
+
+#ifdef USE_NON_SECURE_ADDRESS_MAP
+			/* intentionally empty because WDT30 is hardware fixed to Secure */
+#else
+			wdt30: watchdog@108000 {
+				compatible = "nordic,nrf-wdt";
+				reg = <0x108000 0x620>;
+				interrupts = <264 NRF_DEFAULT_IRQ_PRIORITY>;
+				status = "disabled";
+			};
+#endif
+
+			wdt31: watchdog@109000 {
+				compatible = "nordic,nrf-wdt";
+				reg = <0x109000 0x620>;
+				interrupts = <265 NRF_DEFAULT_IRQ_PRIORITY>;
+				status = "disabled";
+			};
+
+			gpio0: gpio@10a000 {
+				compatible = "nordic,nrf-gpio";
+				gpio-controller;
+				reg = <0x10a000 0x300>;
+				#gpio-cells = <2>;
+				ngpios = <5>;
+				status = "disabled";
+				port = <0>;
+				gpiote-instance = <&gpiote30>;
+			};
+
+			gpiote30: gpiote@10c000 {
+				compatible = "nordic,nrf-gpiote";
+				reg = <0x10c000 0x1000>;
+				status = "disabled";
+				instance = <30>;
+			};
+
+			clock: clock@10e000 {
+				compatible = "nordic,nrf-clock";
+				reg = <0x10e000 0x1000>;
+				interrupts = <270 NRF_DEFAULT_IRQ_PRIORITY>;
+				status = "disabled";
+			};
+		};
+
+		rram_controller: rram-controller@5004e000 {
```

Comment:
It looks this node should be conditional with something like:

```
#ifdef USE_NON_SECURE_ADDRESS_MAP

rram_controller: rram-controller@4004e000 {

#else

rram_controller: rram-controller@5004e000 {
			compatible = "nordic,rram-controller";
			reg = <0x5004e000 0x1000>;
			interrupts = <75 NRF_DEFAULT_IRQ_PRIORITY>;

#endif

			#address-cells = <1>;
			#size-cells = <1>;
			cpuapp_rram: rram@0 {
				compatible = "soc-nv-flash";
...
```

The driver should be available only for secure firmware because this peripheral is accessible only from S mode (removed `compatible` for NS mode). Also, the address for S and NS modes should differ, though in NS mode it's probably not relevant as the driver is unused.